### PR TITLE
Improve Core Info Form Validation

### LINKED ISSUE #481: PaymentProvisioning and BusinessForm - Improve validations for Core Info
We recently made some solid improvements to our address validation in both the PaymentProvisioning and BusinessForm components. 

Let's try to adopt a similar strategy for each step / section of the form. 

Set up specific rules for each field of each schema, chain them with either `required` or `nullable` depending on `allowOptionalFields`, just like in `business-address-schema`.

It might be great to establish a common.ts file inside the schema directory that contains base validations for things like strings, numbers, dates, etc. 🤔 

-------

- For Core Info Step and Core Info in BusinessForm:
- Emulate validation strategy in business-address-schema.ts
- Remove `business_structure` field from form and validation requirement.

### PR DESCRIPTION
**Pull request by colganmeanor**
### Improve Core Info Form Validation

Sorry for the vague PR title - this one does the following:

- Improves validation for core-info form sections for BusinessForm and PaymentProvisioning
- Removes `business_structure` field from both components, removed from validation. 
- Adds validation requirements for Tax ID, adjusts mask value from 11 characters to 9 characters (this is the way it is on formstack, and seems to be the legal format set by the IRS for US companies.)
- renamed `helpers.ts` => `schema-helpers.ts` and moved schema related helpers there, like Regex and a transform function. 

NOTE: this PR will slightly change the DOM layout for both BusinessForm and PaymentProvisioning. `business_structure` select menu has been removed, and some fields have been slightly moved or modified to correct the layout. Will need to note this for our Chromatic runs. 

Links
-----

Closes #481 

Developer considerations
--------------

- On any task
  - [X] Previous unit and e2e tests are passing
  - [x] Perform dev QA

Developer QA steps
--------------------

- [x] Verify that business_structure field should be removed from BusinessForm and PaymentProvisioning (step 1)
- [x] Verify that `tax_id` is now required by default, and has validation rules.
- [x] Verify that validation rules are correctly applied to the following fields: `legal_name`, `phone`, `website_url`, `email`, `doing_business_as`, `business_type`, `industry`, `tax_id`
- [x] Verify that new validations work as expected if `allow-optional-fields` is set to `TRUE`

### CODE REVIEW

File: packages/webcomponents/src/components/business-forms/schemas/business-address-schema.ts [hunk 1]
```diff
@@ -1,40 +1,31 @@
...
 
 const lineOneValidation = string()
   .min(5, 'Address must be at least 5 characters')
   .max(100, 'Address must be less than 100 characters')
-  .matches(/^(?!^\s+$)[a-zA-Z0-9\s,.'-]*$/, 'Invalid characters in address')
-  .transform((value) => {
-    return value === '' ? null : value;
-  })
+  .matches(/^(?!^\s+$)[a-zA-Z0-9\s,.'-]*$/, 'Enter valid address line 1')
+  .transform(transformEmptyString);
```
Comment (TiagoFuelber) [inline packages/webcomponents/src/components/business-forms/schemas/business-address-schema.ts:9]: Nice one!

**Comment by TiagoFuelber** [comment]
Quick question: 

Shouldn't we also remove `business_structure` from the `justifi-business-details` component? 

image

**Comment by colganmeanor** [comment]
> Quick question:
> 
> Shouldn't we also remove `business_structure` from the `justifi-business-details` component?
> 
> image

@TiagoFuelber ohhhhh, yep that makes sense to me. 

@jakemer are you cool with that?

**Review by TiagoFuelber — APPROVED** [review]
Great work! Approved

**Pull request merged**